crates/e2e/src/nodes/mod.rs

```diff
@@ -1,9 +1,113 @@
 pub mod forked_node;
 pub mod local_node;
 
+/// The default node URL that should be used for e2e tests.
 pub const NODE_HOST: &str = "http://127.0.0.1:8545";
 
-#[async_trait::async_trait(?Send)]
-pub trait TestNode {
-    async fn reset(&self);
+/// A blockchain node for development purposes. Dropping this type will
+/// terminate the node.
+pub struct Node {
+    process: Option<tokio::process::Child>,
+}
+
+impl Node {
+    /// Spawns a new node that is forked from the given URL. It also
+    /// impersonates the given `solver` address.
+    pub async fn forked(fork: impl reqwest::IntoUrl) -> Self {
+        Self::spawn_process(&["--port", "8545", "--fork-url", fork.as_str()]).await
+    }
+
+    /// Spawns a new local test net with some default parameters.
+    pub async fn new() -> Self {
+        Self::spawn_process(&[
+            "--port",
+            "8545",
+            "--gas-price",
+            "1",
+            "--gas-limit",
+            "10000000",
+            "--base-fee",
+            "0",
+            "--balance",
+            "1000000",
+            "--chain-id",
+            "1",
+            "--timestamp",
+            "1577836800",
+        ])
+        .await
+    }
+
+    /// Spawn a new node instance using the list of given arguments.
+    async fn spawn_process(args: &[&str]) -> Self {
+        use tokio::io::AsyncBufReadExt as _;
+
+        // Allow using some custom logic to spawn `anvil` by setting `ANVIL_COMMAND`.
+        // For example if you set up a command that spins up a docker container.
+        let command = std::env::var("ANVIL_COMMAND").unwrap_or("anvil".to_string());
+
+        let mut process = tokio::process::Command::new(command)
```

Comment:
Mostly taken from the node implementation in the `driver` test crate. I think sharing this implementation is probably not super useful so I just copied it here.